### DISCUSSION TITLE
How to set the briefcase program's preferences?

### DISCUSSION THREAD

**Discussion by yjlc-pc** [beeware/briefcase#1747] [Q&A]
I can find 'preference' for the briefcase program,but I don't know how to set this.

**Comment by freakboy3742** [comment]
I presume you're asking about the `Preferences` menu item. That's currently a placeholder, provided to indicate that this is an area that we're intending to look at in the future. There isn't any public API to customise that menu item at present.

If you want to use this menu item, you'll need to modify the list of commands to delete the existing Preference menu item, and install your own.

PR #2244 has a prototype of an implementation that allows customisation of the Preferences menu item. It's likely that API, or something close to it, will land in the next couple of months.

**Comment by yjlc-pc** [comment]
Thank you.But how to delete the preference menu item?

**Reply by freakboy3742** [reply]
You need to search the existing `app.commands` definitions for the item that corresponds to the Preferences menu item, and delete that item from the set of commands.

**Reply by yjlc-pc** [reply]
How to search? Can you give me some command?

**Reply by mhsmith** [reply]
The [`app.commands`](https://toga.readthedocs.io/en/stable/reference/api/app.html#toga.App.commands) documentation shows that it's a mutable set. You can iterate through it in a `for` loop, identify the preferences item by its [`text`](https://toga.readthedocs.io/en/stable/reference/api/resources/command.html#toga.Command), then pass it to `app.commands.remove`.

**Comment by reefwing** [comment]
I'm assuming that you mean toga and not briefcase, if that is so then to do this for `Settings...` I used:

```
def startup(self):
        # Other startup stuff...
        
        # Get a reference to the Settings... command
        settings_command = next((cmd for cmd in self.commands if isinstance(cmd, toga.Command) and cmd.text == "Settings\u2026"), None)
        if settings_command is not None:
            settings_command.action = wrapped_handler(settings_command, self.action_settings)
            settings_command.enabled = True
            
def action_settings(self, widget):
    # Display settings window here
    pass
```

Not sure about preferences, I don't see an item for that on macOS running a toga app.

**Reply by freakboy3742** [reply]
The title of the Preferences item is platform dependent; the code you've provided here will work for macOS, but on other platforms the label will be different. However, this general approach will work, though.

**Reply by reefwing** [reply]
That makes sense. It's on my list of things to do to package up an app on all the other platforms, given BeeWare is all about cross platform!
